test/functional/role_test.rb

```diff
@@ -0,0 +1,72 @@
+require File.join(File.dirname(__FILE__), 'test_helper')
+
+describe 'role' do
+  describe 'clone' do
+    before do
+      @cmd = %w(role clone)
+    end
+
+    it 'should print error on missing --id' do
+      params = ['--new-name=zzz']
+
+      expected_result = CommandExpectation.new
+      expected_result.expected_err =
+        ['Could not clone the user role:',
+         "  Missing arguments for 'id'",
+         ''].join("\n")
+      expected_result.expected_exit_code = HammerCLI::EX_USAGE
+
+      api_expects_no_call
+
+      result = run_cmd(@cmd + params)
+      assert_cmd(expected_result, result)
+    end
+
+    it 'should print error on missing --new-name' do
+      params = ['--id=1']
+
+      expected_result = usage_error_result(
+        @cmd,
+        'Option --new-name is required',
+        'Could not clone the user role')
+
+      api_expects_no_call
+
+      result = run_cmd(@cmd + params)
+
+      assert_cmd(expected_result, result)
+    end
+
+    it 'should clone a role by id' do
+      params = ['--id=1', '--new-name=zzz']
+
+      api_expects(:roles, :clone, 'Clone role') do |par|
+        par['id'] == '1' && par['new_name'] == 'zzz'
```

Comment:
I do think this expectation is wrong. Tests fail for me with apidoc export from latest master (with the api counterpart merged). Please make sure the PR contains latest apidoc export.
```
  1) Failure:
role::clone#test_0003_should clone a role by id [/usr/local/rvm/gems/ruby-2.3.0/gems/apipie-bindings-0.0.19/lib/apipie_bindings/api.rb:174]:
unexpected invocation: #<AnyInstance:ApipieBindings::API>.call_action(<Action roles:clone>, {'id' => '1', 'role' => {'name' => 'zzz'}, 'name' => 'zzz'}, {}, {:fake_response => nil})
unsatisfied expectations:
- expected exactly once, not yet invoked: Clone role
  #<AnyInstance:ApipieBindings::API>.call_action(:roles, :clone, &block)

  2) Failure:
role::clone#test_0004_should clone a role by name [/usr/local/rvm/gems/ruby-2.3.0/gems/apipie-bindings-0.0.19/lib/apipie_bindings/api.rb:174]:
unexpected invocation: #<AnyInstance:ApipieBindings::API>.call_action(<Action roles:clone>, {'id' => 1, 'role' => {'name' => 'zzz'}, 'name' => 'zzz'}, {}, {:fake_response => nil})
unsatisfied expectations:
- expected exactly once, not yet invoked: Clone role
  #<AnyInstance:ApipieBindings::API>.call_action(:roles, :clone, &block)
satisfied expectations:
- expected exactly once, invoked once: Attempt find role
  #<AnyInstance:ApipieBindings::API>.call_action(:roles, :index, &block)
```

One more note, not important one, just FYI:
The block param matchers can now be replaced with `.with_params`. The advantage is getting much better error report (diff of params) when the expectation fails. See [testing api expectations docs](https://github.com/theforeman/hammer-cli-foreman/blob/master/doc/testing.md#available-helper-functions) for more details.